feat: add numbersToHexadecimal option to control number to hex conversion

jbetala7 opened the pull request:
## 🎯 Purpose
Resolves #1317 - Adds option to prevent automatic conversion of numbers to hexadecimal format.

## 📋 Changes
- ✅ Added `numbersToHexadecimal` boolean option (default: `true`)
- ✅ Updated `NumberLiteralTransformer` to respect the new option
- ✅ Added CLI support with `--numbers-to-hexadecimal` flag
- ✅ Updated all configuration presets and interfaces
- ✅ Added comprehensive test coverage
- ✅ Updated README documentation with examples

## 🔄 Behavior
- **Default (`true`)**: Maintains existing behavior (backward compatible)
- **New (`false`)**: Keeps numbers in decimal format

## 🧪 Testing
Added test cases covering both scenarios:
- ✅ `numbersToHexadecimal: true` converts to hex
- ✅ `numbersToHexadecimal: false` preserves decimal

## 📚 Documentation
- Updated README with option description and examples
- Added CLI option documentation
- Updated default options section

## 🔧 Usage Examples
```javascript
// JavaScript API
JavaScriptObfuscator.obfuscate(code, {
    numbersToHexadecimal: false
});

// CLI
javascript-obfuscator input.js --numbers-to-hexadecimal false
```
```

This solution will help not just @acemtp, but any other users who prefer to keep their numbers in decimal format for readability or specific requirements!

jbetala7 committed:
feat: add numbersToHexadecimal option to control number to hex conversion

- Add new numbersToHexadecimal option (default: true) to control whether numbers are converted to hexadecimal format
- When true, numbers like 1000 are converted to 0x3e8
- When false, numbers remain in their original decimal format
- Update CLI interface to support --numbers-to-hexadecimal flag
- Add comprehensive documentation with examples
- Update all relevant presets and tests
- Fixes issue where users couldn't disable number to hex conversion

sanex3339 commented:
Please remove package-lock.json and yarn.lock files from this PR, and i will merge it

sanex3339 commented:
I still see yarn.lock conflict

jbetala7 commented:
Rebased on current master and removed the lockfile changes from the PR diff. The PR now only includes the feature/docs/test files; package-lock.json and yarn.lock are no longer modified.

Verification:
- corepack yarn run test:mocha -- --grep numbersToHexadecimal
- git diff --check upstream/master...HEAD
